src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/CrossTarget/AggregateCrossTargetProjectContextProvider.cs

```diff
@@ -241,6 +245,18 @@ private static string GetDisplayName(ConfiguredProject configuredProject, Projec
                 projectData.DisplayName;
         }
 
+        private Task ExecuteWithinLockAsync(Action action) => _gate.ExecuteWithinLockAsync(JoinableCollection, JoinableFactory, action);
+
+        private int _isInitialized;
+
+        private async Task EnsureInitialized()
+        {
+            if (Interlocked.CompareExchange(ref _isInitialized, 1, 0) == 0)
+            {
+                await InitializeAsync();
```

Comment:
Could just return the `Task` here, and `Task.CompletedTask` in the usual case (zero allocation).